Dalamud/Game/Gui/Nameplates/NameplateGui.cs

```diff
@@ -0,0 +1,178 @@
+using System.Linq;
+using System.Runtime.InteropServices;
+
+using Dalamud.Game.ClientState.Objects;
+using Dalamud.Game.ClientState.Objects.Types;
+using Dalamud.Game.Gui.Nameplates.EventArgs;
+using Dalamud.Game.Gui.Nameplates.Model;
+using Dalamud.Game.Text.SeStringHandling;
+using Dalamud.Hooking;
+using Dalamud.IoC.Internal;
+using Dalamud.Memory;
+
+using FFXIVClientStructs.FFXIV.Client.UI;
+
+namespace Dalamud.Game.Gui.Nameplates;
+
+/// <summary>
+/// This class handles interacting with native Nameplate update events and management.
+/// </summary>
+[InterfaceVersion("1.0")]
+[ServiceManager.EarlyLoadedService]
+internal class NameplateGui : IInternalDisposableService, INameplatesGui
+{
+    private readonly GameGui gameGui;
+    private readonly ObjectTable objectTable;
+
+    private readonly NameplateGuiAddressResolver addresses;
+    private readonly IntPtr namePlatePtr;
+
+    private Hook<SetPlayerNameplateDetourDelegate>? setPlayerNameplateDetourHook = null;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NameplateGui"/> class.
+    /// </summary>
+    [ServiceManager.ServiceConstructor]
+    private NameplateGui(TargetSigScanner sigScanner)
+    {
+        // Services
+        this.gameGui = Service<GameGui>.Get();
+        this.objectTable = Service<ObjectTable>.Get();
+
+        // Pointers
+        this.namePlatePtr = this.gameGui.GetAddonByName("NamePlate", 1);
```

Comment:
This is likely to explode when the player logs out and logs back in.

The Addon is re-created when you login.

Reply:
Possible fix includes using AddonLifcycle to listen for the Setup and Finalize events for NamePlate, and use that to manage the pointer.

Reply:
Done!